frappe/desk/search.py

```diff
@@ -275,39 +272,25 @@ def get_std_fields_list(meta, key):
 	return sflist
 
 
-def get_title_field_query(meta):
-	title_field = meta.title_field if meta.title_field else None
-	show_title_field_in_link = (
-		meta.show_title_field_in_link if meta.show_title_field_in_link else None
-	)
-	field = None
-
-	if title_field and show_title_field_in_link:
-		field = f"`tab{meta.name}`.{title_field} as `label`"
-
-	return field
-
+def build_for_autosuggest(res: list[tuple], doctype: str) -> list[dict]:
+	def to_string(parts):
+		return ", ".join(
+			unique(_(cstr(part)) if meta.translated_doctype else cstr(part) for part in parts if part)
+		)
 
-def build_for_autosuggest(res, doctype):
 	results = []
 	meta = frappe.get_meta(doctype)
-	if not (meta.title_field and meta.show_title_field_in_link):
-		for r in res:
-			r = list(r)
-			results.append({"value": r[0], "description": ", ".join(unique(cstr(d) for d in r[1:] if d))})
-
+	if title_field_exists := meta.title_field and meta.show_title_field_in_link:
```

Comment:
```suggestion
	if meta.title_field and meta.show_title_field_in_link:
```

Walrus not required? Since you're not using it anywhere else in the block.